WcaOnRails/db/migrate/20160731000008_update_team_members_with_positions.rb

```diff
@@ -0,0 +1,526 @@
+             select NULL, t.id, u.id, '2003-01-01', NULL, 0, now(), now(), cp.id \
+             from committee_positions cp \
+             join committees c on c.id = cp.committee_id \
+             join teams t on c.id = t.committee_id \
+             cross join users u \
+             where u.delegate_status = 'senior_delegate' \
+             and u.id = 246 \
+             and c.slug = 'wca-delegates-committee' \
+             and t.slug = 'usa-west' \
+             and cp.slug = 'senior-delegate';"
+    execute "insert into team_members\
+             (id, team_id, user_id, start_date, end_date, team_leader, created_at, updated_at, committee_position_id) \
+             select NULL, t.id, u.id, '2003-01-01', NULL, 0, now(), now(), cp.id \
+             from committee_positions cp \
+             join committees c on c.id = cp.committee_id \
+             join teams t on c.id = t.committee_id \
+             cross join users u \
+             where u.senior_delegate_id = 246 \
+             and u.delegate_status = 'delegate' \
+             and c.slug = 'wca-delegates-committee' \
+             and t.slug = 'usa-west' \
+             and cp.slug = 'delegate';"
+    execute "insert into team_members\
+             (id, team_id, user_id, start_date, end_date, team_leader, created_at, updated_at, committee_position_id) \
+             select NULL, t.id, u.id, '2003-01-01', NULL, 0, now(), now(), cp.id \
+             from committee_positions cp \
+             join committees c on c.id = cp.committee_id \
+             join teams t on c.id = t.committee_id \
+             cross join users u \
+             where u.senior_delegate_id = 246 \
+             and u.delegate_status = 'candidate_delegate' \
+             and c.slug = 'wca-delegates-committee' \
+             and t.slug = 'usa-west' \
+             and cp.slug = 'candidate-delegate';"
+  end
+
+  def update_start_dates
+    execute "update team_members tm \
+             set start_date = ifnull(( \
+                   select min(str_to_date(concat(c.year, '-', lpad(c.month, 2, '00'), '-', lpad(c.day, 2, '00')), '%Y-%m-%d')) start_date \
+                   from Competitions c  \
+                   join competition_delegates cd on cd.competition_id = c.id \
+                   join users u on u.id = cd.delegate_id \
+                   where u.id = tm.user_id \
+                   and c.results_posted_at is not null \
+                   group by u.id), now()), \
+                 updated_at = now() \
+             where tm.end_date is null \
+             and   tm.team_id in \
+                   (select t.id \
+                    from teams t \
+                    join committees cm on t.committee_id = cm.id \
+                    where cm.slug = 'wca-delegates-committee');"
+
+    execute "update team_members \
+             set    start_date = now() \
+             where  start_date = '0000-00-00';"
```

Comment:
That's what the SQL statement above does. Then if they haven't delegated a competition yet the start date is set to now().